api/services/user_contact_service.py

```diff
@@ -25,16 +25,32 @@ def init_data(self, jsonData):
     def handle_contact_group(self, jsonData):
         self.init_data(jsonData)
         contact_groups = jsonData["contact"]["groups"]
-        self.mark_user_groups_as_inactive()
+        active_user_groups = models.UserGroup.query.get_by_user_phone(self.user_phone)
+        active_user_group_uuid_list = {}
+
+        for user_group in active_user_groups:
+            active_user_group_uuid_list[
+                str(user_group.group_uuid)
+            ] = models.UserGroup.UserGroupStatus.ACTIVE
+
         for group in contact_groups:
             group_data = models.UserGroup.query.get_unique(
                 group["uuid"], self.user_phone
             )
-            if group_data:
-                self.udpate_group(group_data)
-            else:
+            if (
+                group_data
+                and group_data.status == models.UserGroup.UserGroupStatus.INACTIVE
+            ):
+                self.update_group(group_data)
+            elif not group_data:
                 self.add_group(group)
 
+            if group["uuid"] in active_user_group_uuid_list:
```

Comment:
Removing the user groups that need to be in an active state. This way we will be only left with the groups that are active in our database and should be mark as inactive